packages/cheqd/src/anoncreds/utils/identifiers.ts

```diff
@@ -0,0 +1,50 @@
+import type { ParsedDid } from '@aries-framework/core'
+
+const UUID = '([a-z,0-9,-]{36,36})'
```

Comment:
I think, we can add some static checks for DID or DID URL and raise an error first.
- For UUID validation let's use some specific packages or maybe regexp from UUID RFC cause current UUID regexp is not making the validation.
- If you want to get `method` or `network` or other parts of DID I think it worse to make just split(":") and handle each parts one by one.
- ID in cheqd DID it's not just string from 21 or 22 bytes. Firstly it should be base58 string and the next rule - it should 16 byte after encoding string to array of bytes. 
- About `fragment` or `query` I think it worse to use some `url` parser package and get those values from it or maybe also use official regexps from corresponded RFC.
- The main reason why it would be better - regexps are bad for support in future and for debugging from my point of view.

Reply:
Also, small thing here. Fragment is not placed in URL as `#` it should be encoded to particular code, like `%23`. It's not being sent to the server directly. So, if you want to get the fragment you need to search for `%23` symbol.

Reply:
Resolved, replaced regex validations with proper library validations